### ISSUE TITLE
MUCLight  - chat laggy for message, send and received after 10seconds to members

### ISSUE THREAD

**Issue by vickydhas** [robbiehanson/XMPPFramework#840]
Hello all,

Have anyone implemented the MUC Light  group chat, have any one faced the lag in message sending and receiving.

I have managed to stop the following items to reduce the lag but still not sufficient.

1) Autoping from both  (client & server sides)
2) Disable stream management and autoresume
3) Disable autofetch roster
4) Disable the typing and paused state sending.
5) Disable the Debug mode on both sides of app.

Any suggestions or better guidance is really appreciated.

**Comment by adozenlines** [comment]
If you're seeing this during development it is usually the console logging, turn off all forms or reduce some of your console logging by the framework.

**Comment by vickydhas** [comment]
I tried doing this but less improvements. all the above points if you see the 4, 5 ive mentioned that I have already disabled the logs from Apps.

Also the XMPPLogging.h the following code I modified in the XMPPFramework.

```
#ifndef XMPP_LOGGING_ENABLED
#define XMPP_LOGGING_ENABLED 0
#endif
```

**Comment by adozenlines** [comment]
Hmm are all of your modules running on the same dispatch queue most likely a private queue away from the main queue/thread?

**Comment by vickydhas** [comment]
I got the fix, the Mod_Zero Push code we added in ejabberd to enable push notification was sending push over the main thread and , once spawned to different thread this chat became super fast. Closing this issue with comments so can help someone.

Lesson: If the chat performance is degraded when sending and receiving messages, choose to disable the custom modules created and enabled in ejabberd.cfg file in chat server and check if things improvises.
